### PR TITLE
ADMINUI-2342 Network pool should allow only one type of address, either IPv4 or IPv6.

### PR DESCRIPTION
**Pull request by Arc311**

### CODE REVIEW

File: www/js/views/network-pools-form.js [hunk 1]
```diff
@@ -49,7 +52,16 @@ module.exports = Backbone.Marionette.ItemView.extend({
 
         this.listenTo(this.userInput, 'selected', this.onSelectUser);
         this.listenTo(this.networks, 'sync', function () {
-            this.networks = this.networks.fullCollection;
+            var self = this;
+            this.allNetworks = this.networks.fullCollection;
+            this.allNetworks.forEach(function (network) {
+                if (/:/.test(network.attributes.subnet)) {
```
Comment (dan-luk) [inline www/js/views/network-pools-form.js]: why use regex

File: www/js/views/network-pools-form.js [hunk 2]
```diff
@@ -58,6 +70,19 @@ module.exports = Backbone.Marionette.ItemView.extend({
...
+        this.checkInput();
+        var data = Backbone.Syphon.serialize(this);
+        if (data.networks.length === 1 && (!this.selectedNetworks || this.selectedNetworks.length === 0)) {
+            this.selectedNetworks = data.networks;
+            this.render();
+        } else if (!data.networks) {
+            this.networks = this.allNetworks;
+            this.selectedNetworks = [];
+            this.render();
+        }
```
Comment (dan-luk) [inline www/js/views/network-pools-form.js]: else no render?

File: www/js/views/network-pools-form.js [hunk 3]
```diff
@@ -92,13 +117,22 @@ module.exports = Backbone.Marionette.ItemView.extend({
...
-        if (networkPool.networks) {
-            _.each(networks, function (d) {
-                if (networkPool.networks.indexOf(d.uuid) !== -1) {
-                    d.selected = true;
-                }
+        if (poolNetworks) {
+            var selectedNetwork = _.find(networks, function (network) {
+                return poolNetworks.indexOf(network.uuid) > -1;
             });
+            if(selectedNetwork) {
```
Comment (dan-luk) [inline www/js/views/network-pools-form.js]: space

File: www/js/views/network-pools-form.js [hunk 4]
```diff
@@ -92,13 +117,22 @@ module.exports = Backbone.Marionette.ItemView.extend({
...
+                var isIPv6 = /:/.test(selectedNetwork.subnet);
```
Comment (dan-luk) [inline www/js/views/network-pools-form.js]: why use regex

File: www/js/views/network-pools-form.js [hunk 5]
```diff
@@ -92,13 +117,22 @@ module.exports = Backbone.Marionette.ItemView.extend({
...
+                networks = isIPv6 ? this.IPv6Networks.fullCollection.toJSON() :
+                    this.IPv4Networks.fullCollection.toJSON();
```
Comment (dan-luk) [inline www/js/views/network-pools-form.js]: extract .toJSON()

File: www/js/views/network-pools-form.js (hunk 2 above)
Comment (Arc311) [inline www/js/views/network-pools-form.js]: yep, I need rerender view when select empty and I add network to it or after I remove the last one network from select.

**Comment by dan-luk** [comment]
💥

**Pull request closed without merging**